Tests/WorkspaceTests/PrebuiltsTests.swift

```diff
@@ -0,0 +1,545 @@
+            guard case .download(let fileSystem, let destination) = request.kind else {
+                throw StringError("invalid request \(request.kind)")
+            }
+
+            if request.url == "https://github.com/dschaefer2/swift-syntax/releases/download/600.0.1/\(self.swiftVersion)-manifest.json" {
+                try fileSystem.writeFileContents(destination, data: manifestData)
+                return .okay()
+            } else if request.url == "https://github.com/dschaefer2/swift-syntax/releases/download/600.0.1/\(self.swiftVersion)-MacroSupport-macos_aarch64.zip" {
+                try fileSystem.writeFileContents(destination, data: artifact)
+                return .okay()
+             } else {
+                XCTFail("Unexpected URL \(request.url)")
+                return .notFound()
+            }
+        }
+
+        let archiver = MockArchiver(handler: { _, archivePath, destination, completion in
+            XCTAssertEqual(archivePath.pathString, "/home/user/caches/org.swift.swiftpm/prebuilts/swift-syntax/\(self.swiftVersion)-MacroSupport-macos_aarch64.zip".fixwin)
+            XCTAssertEqual(destination.pathString, "/tmp/ws/.build/prebuilts/swift-syntax/\(self.swiftVersion)-MacroSupport-macos_aarch64".fixwin)
+            completion(.success(()))
+        })
+
+        let workspace = try await MockWorkspace(
+            sandbox: sandbox,
+            fileSystem: fs,
+            roots: [
+                rootPackage
+            ],
+            packages: [
+                swiftSyntax
+            ],
+            prebuiltsManager: .init(
+                httpClient: httpClient,
+                archiver: archiver
+            ),
+            customHostTriple: Triple("arm64-apple-macosx15.0")
+        )
+
+        try await workspace.checkPackageGraph(roots: ["Foo"]) { modulesGraph, diagnostics in
+            XCTAssertTrue(diagnostics.filter({ $0.severity == .error }).isEmpty)
+            let rootPackage = try XCTUnwrap(modulesGraph.rootPackages.first)
+            let macroTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooMacros" }))
+            try checkSettings(macroTarget, usePrebuilt: true)
+            let testTarget = try XCTUnwrap(rootPackage.underlying.modules.first(where: { $0.name == "FooTests" }))
+            try checkSettings(testTarget, usePrebuilt: true)
+        }
+    }
+
+    func testDisabled() async throws {
+        let sandbox = AbsolutePath("/tmp/ws")
+        let fs = InMemoryFileSystem()
+
+        let artifact = Data()
+        let (_, rootPackage, swiftSyntax) = try initData(artifact: artifact, swiftSyntaxVersion: "600.0.1")
+
+        let workspace = try await MockWorkspace(
+            sandbox: sandbox,
+            fileSystem: fs,
+            roots: [
+                rootPackage
+            ],
+            packages: [
+                swiftSyntax
+            ],
+            customHostTriple: Triple("arm64-apple-macosx15.0")
```

Comment:
Question: will the test behavior change if this is removed?

Reply:
Yes, because it'll use the real host triple and I want it to use a mock one.

Reply:
A couple of related comments:

1. can we add a similar test that uses the host triple to ensure we gain extra confidence?  ideally, we would run the same test against all supported Triples to ensure we "don't regress", but it's more difficult creating a parameterized tests in XCTest.
2. Can we add a code comment reminding us to to parameterize this test once it's converted to Swift Testing?

Reply:
Maybe once we get parameterized tests.